runtime-light/coroutine/async-stack-methods.h

```diff
@@ -0,0 +1,36 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2025 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#pragma once
+
+#include "runtime-light/coroutine/async-stack.h"
+
+#include <coroutine>
+#include <utility>
+
+#include "runtime-light/coroutine/coroutine-state.h"
+
+namespace kphp::coro {
+
+using ::CoroutineInstanceState;
+
+/**
+ * The `resume` function is responsible for storing the current synchronous stack frame
+ * in async_stack_root::stop_sync_frame before resuming the coroutine and also creates a new async_stack_root. This allows
+ * capturing one of the stack frames in the synchronous stack trace and jump between parts of the async_stack.
+ */
+[[clang::noinline]] inline void resume(std::coroutine_handle<> handle, async_stack_root* stack_root = nullptr) noexcept {
```

Comment:
Let’s rename `stack_root` to `current_stack_root` or `caller_stack_root`. The current name doesn’t make it clear which stack_root it refers to.